NExtends/Expressions/ExpressionEqualityComparer.cs

```diff
@@ -0,0 +1,239 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NExtends.Expressions
+{
+	/// <summary>
+	/// source : http://stackoverflow.com/questions/283537/most-efficient-way-to-test-equality-of-lambda-expressions
+	/// </summary>
+	public static class ExpressionEqualityComparer
+	{
+		public static bool Eq(
+			LambdaExpression x,
+			LambdaExpression y)
+		{
+			return ExpressionsEqual(x, y, null, null);
+		}
+
+		public static bool Eq<TSource1, TSource2, TValue>(
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of generic parameters in the 'ExpressionEqualityComparer.Eq' method to no more than the 2 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS2436)